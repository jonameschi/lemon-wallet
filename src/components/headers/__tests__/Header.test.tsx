import { ComponentProps } from 'react';
import { Text } from 'react-native';
import { Header } from '@components';
import { fireEvent, render } from '@testing-library/react-native';
import {
  MockNavigation,
  mockNavigationProps,
} from '../../../../__mocks__/MockNavigation';

const renderTree = (props: ComponentProps<typeof Header>) =>
  render(
    <MockNavigation>
      <Header {...props} />
    </MockNavigation>,
  );

describe('Header component', () => {
  it('should render a snapshot with text title', () => {
    const { toJSON } = renderTree({
      title: 'Testing Header',
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render a snapshot - without title', () => {
    const { toJSON } = renderTree({ title: null });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render a snapshot - with ReactNode title', () => {
    const { toJSON } = renderTree({ title: <Text>Testing Header</Text> });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should go back to Home', () => {
    const { getByTestId } = renderTree({ title: 'Testing Header' });
    const button = getByTestId('header-go-back-btn');

    fireEvent.press(button);

    expect(mockNavigationProps.goBack).toHaveBeenCalledTimes(1);
  });
});
