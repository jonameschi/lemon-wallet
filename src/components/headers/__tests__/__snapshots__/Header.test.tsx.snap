// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render a snapshot - with ReactNode title 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 16,
        "justifyContent": "space-between",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="header-go-back-btn"
    >
      <
        height={24}
        width={24}
      />
    </View>
    <Text>
      Testing Header
    </Text>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Header component should render a snapshot - without title 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 16,
        "justifyContent": "space-between",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="header-go-back-btn"
    >
      <
        height={24}
        width={24}
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Header component should render a snapshot with text title 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 16,
        "justifyContent": "space-between",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="header-go-back-btn"
    >
      <
        height={24}
        width={24}
      />
    </View>
    <Text>
      Testing Header
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
